cli/formatter/formatter.go

```diff
@@ -0,0 +1,778 @@
+		if !formatterOpts.TransposeMode {
+			markdownTable = insertStrToStrSlice(markdownTable, 0,
+				constructMarkdownEmptyRow(len(constructHeaderRowForArrs(renderBatch))))
+			markdownTable = insertStrToStrSlice(markdownTable, 1,
+				constructMarkdownHeaderSeparatorRow(len(constructHeaderRowForArrs(renderBatch))))
+		} else {
+			markdownTable = insertStrToStrSlice(markdownTable, 0,
+				constructMarkdownEmptyRow(rowsAmount))
+			markdownTable = insertStrToStrSlice(markdownTable, 1,
+				constructMarkdownHeaderSeparatorRow(rowsAmount))
+		}
+		for _, v := range markdownTable {
+			fmt.Println(v)
+		}
+		fmt.Println()
+	} else if formatterOpts.TableDialect == JiraTableDialect {
+		fmt.Println(t.RenderMarkdown())
+		fmt.Println()
+	} else {
+		fmt.Println(t.Render())
+	}
+
+	return nil
+}
+
+// parseRenderBatch parses render batch.
+func parseRenderBatch(renderBatch []renderNode, formatterOpts *Opts) error {
+	if scalarsOnlyInBatch(renderBatch) {
+		renderSingleColumnTable(renderBatch, formatterOpts)
+
+		return nil
+	}
+
+	if mapsOnlyInBatch(renderBatch) {
+		var renderNodeMaps []map[string]renderNode
+		for _, node := range renderBatch {
+			castedNode, err := castRenderNodeToMap(node)
+			if err != nil {
+				return err
+			}
+			renderNodeMaps = append(renderNodeMaps, castedNode)
+		}
+
+		var renderMapsBatchs = make([][]map[string]renderNode, len(renderNodeMaps))
+		var batchPointer = 0
+		renderMapsBatchs[batchPointer] = append(renderMapsBatchs[batchPointer], renderNodeMaps[0])
+
+		for i := 0; i < len(renderNodeMaps)-1; i++ {
+			if isRenderMapsKeysEqual(renderNodeMaps[i], renderNodeMaps[i+1]) {
+				renderMapsBatchs[batchPointer] = append(
+					renderMapsBatchs[batchPointer],
+					renderNodeMaps[i+1])
+			} else {
+				batchPointer = batchPointer + 1
+				renderMapsBatchs[batchPointer] = append(
+					renderMapsBatchs[batchPointer],
+					renderNodeMaps[i+1])
+			}
+		}
+
+		for _, batch := range renderMapsBatchs {
+			if len(batch) != 0 {
+				if err := renderMapsWithEqualKeys(batch, formatterOpts); err != nil {
+					return err
+				}
+				if formatterOpts.NoGraphics {
+					fmt.Println()
+				}
+			}
+		}
+
+		return nil
+	}
+
+	onlyArrs, singleArrWithArrs, err := arrsOnlyInBatch(renderBatch)
+	if err != nil {
+		return err
+	}
+
+	if onlyArrs {
+		if !singleArrWithArrs {
+			if err := renderArrs(renderBatch, formatterOpts); err != nil {
+				return err
+			}
+		} else {
+			// unpack array with array(s)
+			for _, node := range renderBatch {
+				bytesNode, err := json.Marshal(convertMapIntrfToMapStr(node))
+				if err != nil {
+					return err
+				}
+				iterableNode, err := decodeYamlArr(string(bytesNode))
+				if err != nil {
+					return err
+				}
+				if err := renderArrs(iterableNode, formatterOpts); err != nil {
+					return err
+				}
+			}
+		}
+
+		return nil
+	}
+
+	return nil
+}
+
+// RenderTables renders tables for table/ttable output formats.
+func RenderTables(inputYaml string, formatterOpts *Opts) {
```

Comment:
I suggest do not print the result to the `stdout`, but return as a string. It's easier to test and there are fewer potential problems in the future.
```suggestion
// RenderTables renders tables for table/ttable output formats.
func RenderTables(inputYaml string, formatterOpts *Opts) string {
```

Reply:
Done